plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMigrateProcedure.java

```diff
@@ -224,20 +224,115 @@ public void testMigrateUnsupportedColumnType()
     {
         String tableName = "test_migrate_unsupported_column_type_" + randomNameSuffix();
         String hiveTableName = "hive.tpch." + tableName;
+
+        assertUpdate("CREATE TABLE " + hiveTableName + " AS SELECT tinyint '1' x", 1);
+        assertQuery("SELECT * FROM " + hiveTableName, "SELECT 1");
+
+        assertQueryFails("CALL iceberg.system.migrate('tpch', '%s')".formatted(tableName), "Type not supported for Iceberg: tinyint");
+
+        assertQuery("SELECT * FROM " + hiveTableName, "SELECT 1");
+
+        assertUpdate("DROP TABLE " + hiveTableName);
+    }
+
+    @DataProvider
+    public static Object[][] timestampPrecisionsAndValuesOrc()
+    {
+        return new Object[][] {
+                {HiveTimestampPrecision.MILLISECONDS, "2021-01-01 06:12:12.123"},
+                {HiveTimestampPrecision.MICROSECONDS, "2021-01-01 06:12:12.123456"}
+        };
+    }
+
+    @Test(dataProvider = "timestampPrecisionsAndValuesOrc")
+    public void testMigrateTimestampTypeWithOrc(HiveTimestampPrecision precision, String timestamp)
+    {
+        String tableName = "test_migrate_timestamp_orc" + randomNameSuffix();
+        String hiveTableName = "hive.tpch." + tableName;
         String icebergTableName = "iceberg.tpch." + tableName;
+        Session session = timestampPrecisionSession(precision);
 
-        assertUpdate("CREATE TABLE " + hiveTableName + " AS SELECT timestamp '2021-01-01 00:00:00.000' x", 1);
+        assertUpdate(session, "CREATE TABLE " + hiveTableName + " (x timestamp(" + precision.getPrecision() + ")) WITH (format='ORC')");
+        assertUpdate(session, "INSERT INTO " + hiveTableName + " VALUES timestamp '" + timestamp + "'", 1);
+        assertQuery(session, "SELECT * FROM " + hiveTableName, "VALUES timestamp '" + timestamp + "'");
 
-        assertQueryFails(
-                "CALL iceberg.system.migrate('tpch', '" + tableName + "')",
-                "\\QTimestamp precision (3) not supported for Iceberg. Use \"timestamp(6)\" instead.");
+        assertQuerySucceeds("CALL iceberg.system.migrate('tpch', '" + tableName + "')");
+
+        assertQuery("SELECT * FROM " + icebergTableName, "VALUES timestamp '" + timestamp + "'");
+
+        assertUpdate("DROP TABLE " + icebergTableName);
+    }
+
+    @Test
+    public void testMigrateTimestampMicrosTypeWithAvro()
+    {
+        String tableName = "test_migrate_timestamp_millis_avro" + randomNameSuffix();
+        String hiveTableName = "hive.tpch." + tableName;
+        Session session = timestampPrecisionSession(HiveTimestampPrecision.MICROSECONDS);
+
+        assertUpdate(session, "CREATE TABLE " + hiveTableName + " (x timestamp(6)) WITH (format='AVRO')");
+        assertUpdate(session, "INSERT INTO " + hiveTableName + " VALUES timestamp '2021-01-01 00:00:00.123456'", 1);
+        // Currently Hive maps timestamp(6) to timestamp(millis)
+        assertQuery("SELECT * FROM " + hiveTableName, "VALUES timestamp '2021-01-01 00:00:00.123'");
```

Comment:
This is a matter of what you use for `hive.timestamp_precision` property (which is by default `MILLISECONDS`.

https://github.com/trinodb/trino/blob/b91112a2004997c250b88b4f564dbea0175eca81/testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java#L1002